src/BotChat.tsx

```diff
@@ -1,222 +1,301 @@
+
+        sendMessage: () => {
+            console$.next({text: this.state.console.text, enableSend: false});
+            postMessage(this.state.console.text, this.state.conversation, this.state.userId)
+                .retry(2)
+                .subscribe(
+                    () => {
+                        outgoingMessage$.next({
+                            type: "message",
+                            text: this.state.console.text,
+                            from: {id: 'user'},
+                            timestamp: Date.now().toString()
+                        });
+                        console$.next({
+                            text: "",
+                            enableSend: true
+                        });
+                        autoscroll$.next(true);
+                    },
+                    error => {
+                        console.log("failed to post message");
+                        console$.next({text: this.state.console.text, enableSend: true});
+                    }
+                );
+        },
+
+        sendFile: (files:FileList) => {
+            for (let i = 0, numFiles = files.length; i < numFiles; i++) {
+                const file = files[i];
+                postFile(file, this.state.conversation)
+                    .retry(2)
+                    .subscribe(
+                        () => {
+                            const path = window.URL.createObjectURL(file);
+                            outgoingMessage$.next({
+                                type: "message",
+                                text: this.state.console.text,
+                                from: {id: 'user'},
+                                timestamp: Date.now().toString(),
+                                attachments: [{
+                                    contentType: mimeTypes[path.split('.').pop()],
+                                    contentUrl: path,
+                                    name: 'Your file here'
+                                }]
+                            });
+                        },
+                        error => {
+                            console.log("failed to post file");
+                        }
+                    )
+            }
+        }
+    }
+
+    toggleDebugView() {
+        let newState;
+        if (this.isDebuggerVisible()) {
+            newState = DebugViewState.enabled;
+        } else if (this.isDebuggerEnabled()) {
+            newState = DebugViewState.visible;
+        } else {
+            newState = DebugViewState.disabled;
+        }
+        if (newState !== DebugViewState.visible) {
+            selectedActivity$.next(null);
+        }
+        debugViewState$.next(newState);
+    }
+
+    isDebuggerVisible() {
+        return this.state.debugViewState === DebugViewState.visible;
+    }
+
+    isDebuggerEnabled() {
+        return this.state.debugViewState !== DebugViewState.disabled;
+    }
+
+    render() {
+        return <div className="wc-app">

```

Comment:
I prefer putting "return (" on the first line, and then indent the DIV underneath it so that all the tags line up under each other.